src/_scrapers.py

```diff
@@ -27,24 +30,32 @@ def __init__(self, logger, stop_event, minimum_rating, minimum_votes, goodreads_
             "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36 Edg/111.0.1661.62",
             "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36",
         ]
-        self.firexfox_options = webdriver.FirefoxOptions()
-        self.firexfox_options.add_argument("--no-sandbox")
-        self.firexfox_options.add_argument("--disable-dev-shm-usage")
-        self.firexfox_options.add_argument("--avoid-stats=true")
-        self.firexfox_options.add_argument("--window-size=1280,768")
+        self.firefox_options = webdriver.FirefoxOptions()
+        self.firefox_options.add_argument("--no-sandbox")
+        self.firefox_options.add_argument("--disable-dev-shm-usage")
+        self.firefox_options.add_argument("--avoid-stats=true")
+        self.firefox_options.add_argument("--window-size=1280,768")
         if "linux" in platform.platform().lower():
             display = Display(backend="xvfb", visible=False, size=(1280, 768))
             display.start()
         else:
-            self.firexfox_options.add_argument("--headless")
+            self.firefox_options.add_argument("--headless")
 
-    def get_firexfox_driver(self):
-        firexfox_options = self.firexfox_options
-        firexfox_options.add_argument(f"--user-agent={random.choice(self.user_agents)}")
+    def get_firefox_driver(self):
+        firefox_options = self.firefox_options
+        firefox_options.add_argument(f"--user-agent={random.choice(self.user_agents)}")
+        self.diagnostic_logger.info(f"Platform: {platform.platform()}")
         if "linux" in platform.platform().lower():
-            return webdriver.Firefox(options=firexfox_options, service=FireFoxService(GeckoDriverManager().install()))
-        else:
-            return webdriver.Firefox(options=firexfox_options)
+            if "aarch64" in platform.platform().lower():
+                return webdriver.Firefox(
+                    options=firefox_options,
+                    service=FireFoxService(executable_path="/usr/bin/geckodriver"),
+                )
+            return webdriver.Firefox(
+                options=firefox_options,
+                service=FireFoxService(GeckoDriverManager()).install(),
+            )
+        return webdriver.Firefox(options=firefox_options)

```

Comment:
I think it's better to check `if linux and aarch64`, then`elif linux`, and use `else` for everything else.